tools/ipapy2/macros.xml

```diff
@@ -1,3 +1,27 @@
 <macros>
     <token name="@TOOL_VERSION@">1.3.0</token>
+
+    <xml name="citations">
+        <citations>
+        <citation type="doi">10.1021/acs.analchem.9b02354</citation>
+        <citation type="doi">10.1093/bioinformatics/btad455</citation>
+        </citations>
+    </xml>
+
+    <xml name="ionisation">
+        <param label="ionisation" name="ionisation" type="select" help="ionisation. positive +1 or negative -1">
+            <option value="positive">positive</option>
+            <option value="negative">negative</option>
+        </param>
+    </xml>
+
+    <xml name="ncores">
+        <param label="number of cores" name="ncores" type="integer" value="1" help="number of cores">
+        </param>
+    </xml>
```

Comment:
This parameter has to be provided by Galaxy using the `GALAXY_SLOTS` environment variable - you can have a look at the `recetox-aplcms align features` wrapper to see how it can be used.

Reply:
This was also partially resolved in https://github.com/RECETOX/galaxytools/pull/474/commits/04166cde650d100dbc94873bd90959ee9c818b0a, but it still needs to be fixed in some of the tools.